src/inspect_ai/model/_providers/openai.py

```diff
@@ -160,14 +182,25 @@ def __init__(
                     os.environ.get("OPENAI_API_VERSION", "2025-03-01-preview"),
                 )
 
-            self.client: AsyncAzureOpenAI | AsyncOpenAI = AsyncAzureOpenAI(
-                api_key=self.api_key,
-                api_version=api_version,
-                azure_endpoint=base_url,
-                http_client=http_client,
-                timeout=client_timeout if client_timeout is not None else NOT_GIVEN,
-                **model_args,
-            )
+            # Use managed identity if available, otherwise API key
+            if self.token_provider:
+                self.client: AsyncAzureOpenAI | AsyncOpenAI = AsyncAzureOpenAI(
```

Comment:
Here we'll want to create a `dict` that has all of the shared arguments and then just call the two functions with  `azure_ad_token_provider` and `api_key`  along with the shared args (e.g. 

```python
client_args = dict(
    api_version=api_version,
    azure_endpoint=base_url,
    http_client=http_client,
    timeout=client_timeout if client_timeout is not None else NOT_GIVEN)
    | model_args
)

AsyncAzureOpenAI(api_key=self.api_key, **client_args)
```